src/persist/event_store.rs

```diff
@@ -69,18 +69,18 @@ fn test_source_of_truth() {
 }
 
 /// Storage engine using a database backing.
+///
 /// This defaults to an event-sourced store (i.e., events are the single source of truth),
 /// but can be configured to be aggregate-sourced or use snapshots when a large number of events
 /// are associated with a single aggregate instance.
-///
 pub struct PersistedEventStore<R, A>
 where
     R: PersistedEventRepository,
     A: Aggregate + Send + Sync,
 {
     repo: R,
     storage: SourceOfTruth,
-    event_upcasters: Option<Vec<Box<dyn EventUpcaster>>>,
```

Comment:
this is needless indirection, and creates two redundant, semantically equivalent states (`Option::None` + `Option::Some(vec![])`)